sycamore/tests/unit/transforms/test_regex_replace.py

```diff
@@ -0,0 +1,54 @@
+import ray.data
+
+from sycamore.data import Document, Element
+from sycamore.transforms.regex_replace import RegexReplace
+from sycamore.plan_nodes import Node
+
+
+class FakeNode(Node):
+    def __init__(self, doc: dict):
+        self.doc = doc
+
+    def execute(self) -> ray.data.Dataset:
+        return ray.data.from_items([self.doc])
+
+
+class TestRegexReplace:
+    doc = Document(
+        {
+            "doc_id": "doc_id",
+            "type": "pdf",
+            "text_representation": "foobar",
+            "binary_representation": None,
+            "parent_id": None,
+            "properties": {"path": "/docs/foo.txt", "title": "bar"},
+            "elements": [
+                {
+                    "type": "UncategorizedText",
+                    "text_representation": " \t foo   \t\t\n\t \t\n\n \t   bar\t  ",  # noqa: E501
+                },
+            ],
+        }
+    )
+
```

Comment:
Can you add tests for incorrect arguments?
1. Tolerate empty list and do nothing (maybe print a warning, not sure)
2. Not a list, some wrong things in the list (e.g. string rather than pair)
3. Not pairs in the list